### PR TITLE
Pricers for overnight in-arrears cap/floor. 

### PR DESCRIPTION
**Pull request by marc-henrard**
Pricers for caplet/floorlet period on in-arrears overnight rates (Asian type).
Pricer for SABR data and volatility based data.

### CODE REVIEW

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrOvernightInArrearsCapletFloorletPeriodPricer.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+/**
+ * Pricer for in-arrears caplets and floorlets (Asian style options) in the SABR with effective parameters approach.
+ */
+public class SabrOvernightInArrearsCapletFloorletPeriodPricer {
+
+  /**
+   * The default function for Asian in-arrears effective parameters.
+   */
+  private final SabrInArrearsVolatilityFunction sabrInarrearsFunction;
```
Comment (ChristianOG) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrOvernightInArrearsCapletFloorletPeriodPricer.java:41]: very minor: sabrInarrearsFunctions -> sabrInArrearsFunction

**Review by ChristianOG — COMMENTED** [review]
Couldn't find anything serious issues. A few comments.

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrOvernightInArrearsCapletFloorletPeriodPricer.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+      SabrParametersIborCapletFloorletVolatilities sabrVolatilities) {
+
+    Currency currency = period.getCurrency();
+    if (!ratesProvider.getValuationDate().isBefore(period.getEndDate())) {
+      return PointSensitivityBuilder.none();
+    }
+    OvernightCompoundedRateComputation onComputation = period.getOvernightRate();
+    LocalDate startDate = onComputation.getStartDate();
+    LocalDate endDate = onComputation.getEndDate();
+    double startTime = sabrVolatilities.relativeTime(startDate.atStartOfDay(ZoneId.of("Europe/Brussels"))); // xxx
```
[the remark this answers is not in the document]
Comment (marc-henrard) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrOvernightInArrearsCapletFloorletPeriodPricer.java:203]: In the standard LIBOR caplet, the fixing is done at 11:00 and the time is recorded in the index description. Here we don't have an exact time for the fixing, but the signature for the "relativeTime" method is a ZonedDateTime, while here we just have a LocalDate. We have always given the possibility to measure the relative time to actual time and not only dates.
This is not very pretty, but I don't know how to do it in another way.

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrOvernightInArrearsCapletFloorletPeriodPricer.java (hunk 1 above)
Comment (marc-henrard) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SabrOvernightInArrearsCapletFloorletPeriodPricer.java:41]: done

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityOvernightInArrearsCapletFloorletPeriodPricer.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2021 - present by OpenGamma Inc. and the OpenGamma group of companies
```
[the remark this answers is not in the document]
Comment (marc-henrard) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityOvernightInArrearsCapletFloorletPeriodPricer.java:2]: Done

**Review by ChristianOG — APPROVED** [review]
Couldn't find anything serious issues. A few comments.

**Pull request merged**